tests/ignite/metrics/nlp/test_bleu.py

```diff
@@ -28,117 +29,185 @@ def test_wrong_inputs():
+
+        bleu.update((candidates, references))
         assert reference == bleu.compute()
 
 
-@pytest.mark.parametrize(
-    "candidate, references",
-    [
-        ([["a", "a", "a", "b", "c"]], [[["a", "b", "c"], ["a", "a", "d"]]]),
-        corpus.sample_1,
-        corpus.sample_2,
-        corpus.sample_3,
-        corpus.sample_4,
-    ],
-)
-def test_corpus_bleu_nltk_smooth2(candidate, references):
+@pytest.mark.parametrize(*parametrize_args)
+def test_macro_bleu_smooth1(candidates, references):
+    _test_smooth(candidates, references, "macro")
+
+
+@pytest.mark.parametrize(*parametrize_args)
+def test_micro_bleu_smooth1(candidates, references):
+    _test_smooth(candidates, references, "micro")
+
+
+def _test_bleu_nltk_smooth2(candidates, references, average):
     for i in range(1, 8):
         weights = tuple([1 / i] * i)
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            reference = corpus_bleu(
-                references, candidate, weights=weights, smoothing_function=SmoothingFunction().method2
-            )
-        bleu = Bleu(ngram=i, smooth="nltk_smooth2")
-        assert reference == bleu._corpus_bleu(references, candidate)
-        bleu.update((candidate, references))
-        assert reference == bleu.compute()
+        bleu = Bleu(ngram=i, smooth="nltk_smooth2", average=average)
 
+        if average == "macro":
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                reference = sentence_bleu(
+                    references[0], candidates[0], weights=weights, smoothing_function=SmoothingFunction().method2
+                )
+            assert reference == bleu._sentence_bleu(references[0], candidates[0])
 
-@pytest.mark.parametrize(
-    "candidate, references",
-    [
-        ([["a", "a", "a", "b", "c"]], [[["a", "b", "c"], ["a", "a", "d"]]]),
-        corpus.sample_1,
-        corpus.sample_2,
-        corpus.sample_3,
-        corpus.sample_4,
-    ],
-)
-def test_corpus_bleu_smooth2(candidate, references):
+        elif average == "micro":
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                reference = corpus_bleu(
+                    references, candidates, weights=weights, smoothing_function=SmoothingFunction().method2
+                )
+            assert reference == bleu._corpus_bleu(references, candidates)
+
+        bleu.update((candidates, references))
+        assert pytest.approx(reference) == bleu.compute()
+
+
+@pytest.mark.parametrize(*parametrize_args)
+def test_macro_bleu_nltk_smooth2(candidates, references):
+    _test_bleu_nltk_smooth2(candidates, references, "macro")
+
+
+@pytest.mark.parametrize(*parametrize_args)
+def test_micro_bleu_nltk_smooth2(candidates, references):
+    _test_bleu_nltk_smooth2(candidates, references, "micro")
+
+
+def _test_bleu_smooth2(candidates, references, average):
```

Comment:
Same comment than above. 

Careful, Line150 range(1, 3) while range(1, 8) in the method above. It misses an extra argument for the size (default 8).